dotnet/src/SemanticKernel.Abstractions/IKernel.cs

```diff
@@ -186,8 +187,17 @@ public interface IKernel
     /// <returns>Instance of T</returns>
     T GetService<T>(string? name = null) where T : IAIService;
 
-    #region Obsolete
+    /// <summary>
+    /// Used for registering a pre-run event handler.
+    /// </summary>
+    event EventHandler<KernelRunningEventArgs> Running;
```

Comment:
nit: Maybe something like OnExecuting and OnExectued?

Reply:
I did this thinking in the RunAsync, should we use ExecuteAsync then ?

Reply:
The On* naming, IMO would typically be for the handler (not the event itself)
`     kernel.Running += this.OnRunning`

We probably do need event names that better indicate that this will be run before and after each step (not just once for the whole pipeline). This is a tricky one... thinking of recommendations.

- FunctionStarting // FunctionCompleted
- StepRunning // StepCompleted
...

Reply:
Right, as discussed, I'm going to use `FunctionInvoking` and `FunctionInvoked` as those events, which better fits the current behavior and Function.InvokeAsync method.